lib/templateFactory.js

```diff
@@ -6,6 +6,9 @@ const compareVersions = require('compare-versions');
 const schema = require('screwdriver-data-schema');
 let instance;
 
+const TEMPLATE_NAME_REGEX = schema.config.regex.FULL_TEMPLATE_NAME;
+const TEMPLATE_NAME_MATCH = 1;
```

Comment:
It can also be written:

```
[ fullText, templateName, versionOrTag ] = VERSION_REGEX.exec(config.version);
```

It removes the need for a magic number and it still fulfills the need to pull out the necessary RegEx matches.